### PR TITLE
WIP: Support search using POST

### PR DESCRIPTION
**Pull request by kennethmyhra**
PR for issue #479.
Added methods for executing search using POST
Can be tested by using end point at http://nde-fhir-ehelse.azurewebsites.net/fhir

### CODE REVIEW

File: src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs [hunk 1]
```diff
@@ -118,6 +127,17 @@ private static void setBodyAndContentType(HttpWebRequest request, Resource data,
...
+            else if (isSearch)
+            {
+                string bodyParameters = null;
+                foreach(Parameters.ParameterComponent parameter in ((Parameters)data).Parameter)
+                {
+                    if (!string.IsNullOrEmpty(bodyParameters)) bodyParameters += "&";
+                    bodyParameters += $"{parameter.Name}={parameter.Value}";
+                }
+                body = Encoding.UTF8.GetBytes(bodyParameters);
+                request.ContentType = "application/x-www-form-urlencoded";
```
Comment (wmrutten) [inline src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs]: Doesn't the contenttype "x-www-form-urlencoded" imply that the parameters need to be url-encoded?

Comment (kennethmyhra) [reply src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs]: True, committed fix

Comment (wmrutten) [reply src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs]: Thanks, great job. Would be really awesome if you could also add a unit test for some special characters like <&>"'äëïo€ etcetera. Don't know if we do this for GET already, but I think we should. However that's another issue.

**Comment by kennethmyhra** [comment]
Added test cases per request:
* FormUrlEncoding (Special characters)
* Method == POST
* ContentType == application/x-www-form-urlencoded
* Url Suffix == _search

 @wmrutten see commit 381a5d2

File: src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs [hunk 2]
```diff
@@ -53,9 +54,17 @@ internal static class EntryToHttpExtensions
             {
                 request.Headers["Prefer"] = bodyPreference == Prefer.ReturnMinimal ? "return=minimal" : "return=representation";
             }
-
+            
             if (entry.Resource != null)
-                setBodyAndContentType(request, entry.Resource, format, CompressRequestBody, out body);
+            {
+                bool isSearch =
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs]: The name of this variable put me on the wrong foot.  Shouldn't this be called  `bools isSearchUsingPost`?

Comment (kennethmyhra) [reply src/Hl7.Fhir.Core/Rest/EntryToHttpExtensions.cs]: Agree, maybe lose the is-part altogether and call it searchUsingPost.

**Comment by ewoutkramer** [comment]
Hi Kenneth....I applied your changes to STU3 and did some updates to make it compile....

Could you review my changes so we can have your code in STU3 too?

https://github.com/ewoutkramer/fhir-net-api/pull/530

**Comment by kennethmyhra** [comment]
Will do

**Pull request merged**